sympy/stats/stochastic_process.py

```diff
@@ -0,0 +1,146 @@
+from sympy import Symbol, Indexed, sympify, Basic, S
+from sympy.stats.rv import RandomSymbol, _value_check,  NamedArgsMixin
+from sympy.stats.joint_rv import JointPSpace
+from sympy.core.compatibility import string_types
+
+class StochasticRandomSymbol(RandomSymbol):
+
+    def __getitem__(self, time):
+        if not isinstance(self.pspace, StochasticProcess):
+            raise ValueError("Currently %s process is not supported by SymPy."
+                                %(str(self.pspace)))
+        if time not in self.pspace.index_set:
+            raise ValueError("The stochastic process doesn't have %s in its "
+                                " index set"%(time))
+        return Indexed(self, time)
+
+    @property
+    def process(self):
+        return self.pspace.process
+
+class StochasticProcess(JointPSpace):
+
+    def __new__(cls, sym, process=None):
+        if isinstance(sym, string_types):
+            sym = Symbol(sym)
+        if not isinstance(sym, Symbol):
+            raise ValueError("The process name should be a String or Symbol.")
+        if process == None:
+            process = AbstractStochasticProcess
+        return Basic.__new__(cls, sym, process)
+
+    @property
+    def index_set(self):
+        return self.process.index_set
+
+    @property
+    def state_space(self):
+        return self.process.state_space
+
+    @property
+    def symbol(self):
+        return self.args[0]
+
+    @property
+    def process(self):
+        return self.args[1]
+
+    @property
+    def value(self):
+        return StochasticRandomSymbol(self.symbol, self)
+
+    def joint_dist(self, n):
+        return NotImplementedError()
```

Comment:
You need to raise an exception here. So, change it to `raise NotImplementedError()`.

Reply:
Isn't it already `raise NotImplementedError()`?

Reply:
Well that's just a prototype, not a big deal.